GoogleForCreators/web-stories-wp
# Design System: Hierarchical Taxonomy Component

## Context

Adding some taxonomy components to the design system. [Designs](https://www.figma.com/file/1YM3jA9h4QGc4xLdyaXIJS/Stories-Sprint-1.7?node-id=9721%3A383439)

## Summary

Adds a hierarchical input component to the design system.

## Relevant Technical Choices

The data structure and tree building was inspired from [gutenberg](https://github.com/WordPress/gutenberg/blob/3f9968e2815cfb56684c1acc9a2700d8e4a02726/packages/editor/src/utils/terms.js#L13-L40). Keeping the options flat makes it easy to find options when needing to update checked values.

## To-do

n/a

## User-facing changes

None - only in storybook

![input](https://user-images.githubusercontent.com/22185279/132764703-f6e7f111-18fb-4096-b3e9-27180dc4ce23.gif)

|||
|--|--|
|![image](https://user-images.githubusercontent.com/22185279/132764693-b7934f9c-d27f-4dce-865a-fc31d143c48f.png)|![image](https://user-images.githubusercontent.com/22185279/132764779-ca4926d9-bd3e-42c4-93dd-b907e09758eb.png)|

## Testing Instructions






This PR can be tested by following these steps:

1. Load up storybook `$ npm run storybook`
2. Go to `Stories Editor/Components/Form/Hierarchical`
3. Play with the component. Should be keyboard accessible and work with voiceover/chromevox/screen readers similar to the gutenberg editor.
4. When testing accessibility: after typing the input should announce `n results found` 👍 

## Reviews

### Does this PR have a security-related impact?

no

### Does this PR change what data or activity we track or use?

no

### Does this PR have a legal-related impact?

no

## Checklist




---



Fixes #8836
